Document string context (#8595)
NixOS/nix: doc/manual/src/glossary.md

  [output path]: #gloss-output-path

> - [derived path]{#gloss-derived-path}

Author: I have come to disagree with this part. It only seems wrong because our old habit of confusing *store path* and *store object*. The store object itself is derivable, but the path *itself* already derived: derived by adding `^...` to the end as needed.

Happy to go with another name, just don't want that regresses on disentangling *store paths* and *store object*.

> - [derived path]{#gloss-derived-path}

Author: How about "Deriving Path"? A "deriving path" is a path for deriving, just as a "divining rod" is a rod for divining.

> - [derived path]{#gloss-derived-path}

Reviewer: I agree deriving sounds better. Maybe @fricklerhandwerk has thoughts?

> - [derived path]{#gloss-derived-path}

Author: @fricklerhandwerk said on call that it sounded good to him too.

>   An [store]-level expression (not Nix language expression) that denotes a [store object].

Author: ```suggestion
  A [store]-level expression (not Nix language expression) that denotes a [store object].
```

  There are two forms:

  - *constant*: A constant derived path is just a [store path]

  - *output*: An output derived path is a pair of a [store path] to a [derivation] and an [output] name.

>   Derived paths are necessary because in general, we do not know what the [output path] path of an [output] will buntil it is [realise]d.
>   We need a way to refer to unrelised outputs in order to be able to tell Nix to realise them!

Author: ```suggestion
  Derived paths are necessary because in general, we do not know what the [output path] path of an [output] will be until it is [realise]d.
  Derived paths provide a way to refer to unrealised outputs in order to be able to tell Nix to realise them.
```

  [output path]: #gloss-output-path

> - [deriving path]{#gloss-derived-path}

Reviewer 2: ```suggestion
- [deriving path]{#gloss-deriving-path}
```

Otherwise the anchor link in the new page would be broken.

>   A [store]-level expression (not Nix language expression) that denotes a [store object].

Reviewer 3: ```suggestion
  Deriving paths are a way to refer to [store objects][store object] that are not yet [realised][realise].
  This is necessary because, in general and particularly for [content-addressed derivations][content-addressed derivation], the [output path] of an [output] is not known in advance.
```

I really struggle with "store-level expression". This is just very confusing, even if one could argue there is exists some sort of "store language".

  There are two forms:

>   - *constant*: A constant deriving path is just a [store path]

Reviewer 3: ```suggestion
  - *constant*: just a [store path]
```

These are many words that add no information.

It's not wrong to have "constant deriving path" in a sentence somewhere, but we'll likely find more opportunities for that.

>     It can be made valid by copying it into the store, from the evaluator, command line interface or another store.

Reviewer 3: ```suggestion
    It can be made [valid][validity] by copying it into the store: from the evaluator, command line interface or another store.
```

>   - *output*: An output deriving path is a pair of a [store path] to a [derivation] and an [output] name.

Reviewer 3: ```suggestion
  - *output*: a pair of a [store path] to a [derivation] and an [output] name
```

>   Deriving paths are necessary because in general, we do not know what the [output path] path of an [output] will be until it is [realise]d.
>   Deriving paths provide a way to refer to unrealised outputs in order to be able to tell Nix to realise them.

Reviewer 3: ```suggestion
```

Not needed when used in the introduction.
